### DISCUSSION TITLE
Student teams gaining access to other team's forks

### DISCUSSION THREAD

**Discussion by cameronedmond** [community/community#123872] [GitHub Education]
### Hi everyone,

Hi there!

We are using Github Classroom's new fork-based method for handling student repositories. This has worked well this semester with a couple of hiccups as we navigate the system. However, we have encountered a major problem for our end-of-semester group assignment. 

Students self-enroll in pairs and create teams in the Github Classroom when accepting the same starter code. The starter code repo is private.

We have noticed students are automatically given read access to every other team's fork (as well as the parent repo). This creates security concerns.

Currently, the only solution we have found is to:
1. Remove all additional teams from each repo.
2. Give teams admin access to their repos so they can do this removal themselves.

This seems like a process we will need to repeat until all students have created their teams and accepted the assignment.

I have reviewed the template project my colleague set-up and the github classroom assignment, and I can't find anything different from how we set-up our other assignments for this unit. The only differences is that it is a group assignment, and we were still setting all the other ones to template repos.

Is this a known issue, and is there a more tenable solution than manually removing access?

**Comment by nplatis** [comment]
I have given several assignments using Classroom since February. Starting from one I gave around the start of May, all student teams are having read access to all repos of the assignment. The template is read-only.

This is clearly wrong!
